### PR TITLE
storage: auto-generate proto to md

### PR DESCRIPTION
**Pull request by glimchb**
Signed-off-by: Boris Glimcher <Boris.Glimcher@emc.com>

### CODE REVIEW

File: storage/Makefile [hunk 1]
```diff
@@ -4,3 +4,4 @@
 
 all:
 	docker run -v ${PWD}:/defs namely/protoc-all -d proto -l go -o ./proto/  --go-source-relative
+	docker run --rm -v ${PWD}/proto/:/out -v ${PWD}/proto:/protos pseudomuto/protoc-gen-doc --doc_opt=markdown,autogen.md
```
Comment (mestery) [inline storage/Makefile:7]: Please wrap `${PWD}` with brackets so variable expansion works ok, like this: `"${PWD}"`.

File: storage/proto/autogen.md [hunk 2]
```diff
@@ -0,0 +1,1007 @@
+# Protocol Documentation
```
Comment (mestery) [inline storage/proto/autogen.md:1]: Did you auto generate this file and you're checking it in?

**Review by mestery — CHANGES_REQUESTED** [review]
Some changes to ensure variable expansion works ok in all cases.

File: storage/proto/autogen.md (hunk 2 above)
Comment (glimchb) [reply storage/proto/autogen.md:1]: yes

File: storage/Makefile (hunk 1 above)
Comment (glimchb) [reply storage/Makefile]: fixed

**Review by mestery — APPROVED** [review]
thanks!

**Pull request merged**